flink-runtime/src/test/java/org/apache/flink/runtime/jobmanager/JobManagerProcessUtilsTest.java

```diff
@@ -42,19 +49,37 @@
 	private static final MemorySize TOTAL_FLINK_MEM_SIZE = MemorySize.parse("1280m");
 	private static final MemorySize TOTAL_PROCESS_MEM_SIZE = MemorySize.parse("1536m");
 
+	@Rule
+	public final TestLoggerResource testLoggerResource = new TestLoggerResource(JobManagerFlinkMemoryUtils.class, Level.WARN);
+
 	public JobManagerProcessUtilsTest() {
 		super(JM_PROCESS_MEMORY_OPTIONS, JM_LEGACY_HEAP_OPTIONS, JobManagerOptions.TOTAL_PROCESS_MEMORY);
 	}
 
 	@Test
 	public void testConfigJvmHeapMemory() {
-		MemorySize jvmHeapSize = MemorySize.parse("50m");
+		MemorySize jvmHeapMemory = MemorySize.parse("50m");
 
 		Configuration conf = new Configuration();
-		conf.set(JobManagerOptions.JVM_HEAP_MEMORY, jvmHeapSize);
+		conf.set(JobManagerOptions.JVM_HEAP_MEMORY, jvmHeapMemory);
+
+		JobManagerProcessSpec jobManagerProcessSpec = JobManagerProcessUtils.processSpecFromConfig(conf);
+		assertThat(jobManagerProcessSpec.getJvmHeapMemorySize(), is(jvmHeapMemory));
+		MatcherAssert.assertThat(
+			testLoggerResource.getMessages(),
+			hasItem(containsString("The configured or derived JVM heap memory size (50.000mb (52428800 bytes)) is less than its recommended minimum value (128 mb)")));
```

Comment:
I think it would be better to also use `jvmHeapSize.toHumanReadableString()` and `JobManagerOptions.MIN_JVM_HEAP_SIZE.toHumanReadableString()` in test so that we do not need to change hard coded values if they change.

Btw, it looks like we forgot to add `toHumanReadableString` for `JobManagerOptions.MIN_JVM_HEAP_SIZE` in `verifyJvmHeapSize()`.